dyno-lock/src/test/java/com/netflix/conductor/dynolock/lock/DynoLockTest.java

```diff
@@ -0,0 +1,278 @@
+        properties = mock(DynoLockProperties.class);
+
+        host =
+                new HostBuilder()
+                        .setHostname("localhost")
+                        .setIpAddress("127.0.0.1")
+                        .setDatastorePort(REDIS_PORT)
+                        .setPort(REDIS_PORT)
+                        .setRack(REDIS_RACK)
+                        .setStatus(Host.Status.Up)
+                        .createHost();
+        tokenMapSupplier = new TokenMapSupplierImpl(host);
+
+        HostSupplier hostSupplier = () -> Collections.singletonList(host);
+
+        when(properties.getHosts()).thenReturn("dyno1:8102:us-east-1c");
+        when(properties.getAppId()).thenReturn("conductor");
+        when(properties.getClusterName()).thenReturn("dyn_o_mite");
+        when(properties.getTimeout()).thenReturn(50L);
+        when(properties.getAvailabilityZone()).thenReturn("us-east-1c");
+        when(properties.getDataCenterRegion()).thenReturn("us-east-1");
+        when(properties.getMaxConnectionsPerHost()).thenReturn(10);
+
+        startRedisServer();
+
+        dynoLock =
+                getDynoLock(
+                        properties.getAppId(),
+                        properties.getClusterName(),
+                        hostSupplier,
+                        null,
+                        getCPConfig(),
+                        properties.getTimeout(),
+                        TimeUnit.MILLISECONDS,
+                        null);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        redisServer.stop();
+    }
+
+    private static void startRedisServer() throws IOException {
+        redisServer = new RedisServer(REDIS_PORT);
+        if (redisServer.isActive()) {
+            redisServer.stop();
+        }
+        redisServer.start();
+    }
+
+    private static ConnectionPoolConfigurationImpl getCPConfig() {
+        final ConnectionPoolConfigurationImpl connectionPoolConfiguration =
+                new ConnectionPoolConfigurationImpl(REDIS_RACK);
+        connectionPoolConfiguration.withTokenSupplier(tokenMapSupplier);
+        connectionPoolConfiguration.setLocalRack(REDIS_RACK);
+        connectionPoolConfiguration.setLocalDataCenter(REDIS_DATACENTER);
+        connectionPoolConfiguration.setConnectToDatastore(true);
+        return connectionPoolConfiguration;
+    }
+
+    private static DynoLock getDynoLock(
+            String appName,
+            String clusterName,
+            HostSupplier hostSupplier,
+            TokenMapSupplier tokenMapSupplier,
+            ConnectionPoolConfigurationImpl cpConfig,
+            long timeout,
+            TimeUnit timeoutUnit,
+            EurekaClient eurekaClient) {
+        DynoOPMonitor opMonitor = new DynoOPMonitor(appName);
+        ConnectionPoolMonitor cpMonitor = new DynoCPMonitor(appName);
+
+        DynoJedisUtils.updateConnectionPoolConfig(
+                cpConfig, hostSupplier, tokenMapSupplier, eurekaClient, clusterName);
```

Comment:
from where the hostSupplier is configured?

Reply:
what will happen to hostsupplier when we have more than one host configured in the RedisProperties file?